server/message_hooks.go

```diff
@@ -119,14 +119,18 @@ func (p *Plugin) MessageHasBeenPosted(_ *plugin.Context, post *model.Post) {
 			return
 		}
 
-		chatMembersSpanPlatforms, appErr := p.ChatMembersSpanPlatforms(members)
-		if appErr != nil {
-			p.API.LogWarn("Failed to check if chat members span platforms", "error", appErr.Error(), "post_id", post.Id, "channel_id", post.ChannelId)
-			return
-		}
+		isSelfPost := len(members) == 1
+		chatMembersSpanPlatforms := false
+		if !isSelfPost {
```

Comment:
If selfPost...allow to sync, setting `chatMembersSpanPlatforms := false`. This variable is passed to `SendChat` below. `SendChat` uses this variable to determine whether or not to display the "connect" message if the user is not connected.